### PR TITLE
Adding httpRequest host function

### PR DESCRIPTION
**Pull request by blinktag**
* Add `httpRequest()` host function to support httprequest.wasm
* Prefixing OperationType constants
* Adding timeouts for pipeline and step

### CODE REVIEW

File: function.go [hunk 1]
```diff
@@ -137,18 +143,127 @@ func createWASMInstance(wasmBytes []byte) (api.Module, error) {
...
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to instantiate wasm module")
 	}
 
 	return mod, nil
 }
+
+// httpRequest is function that is exported to and called from the Rust WASM module
+func httpRequest(_ context.Context, module api.Module, ptr, length int32) int32 {
+	//return httpResponse(module, http.StatusInternalServerError, "omgwtf", nil)
```
Comment (dselans) [inline function.go:174]: Remove

File: function.go [hunk 2]
```diff
@@ -137,18 +143,127 @@ func createWASMInstance(wasmBytes []byte) (api.Module, error) {
...
+
+	// Apply terminator to end of response
+	out = append(out, 166, 166, 166)
+
+	alloc := module.ExportedFunction("alloc")
+
+	allocRes, err := alloc.Call(context.Background(), uint64(len(out)))
+	if err != nil {
+		// TODO: do we need to dealloc here?
+		panic(err)
```
Comment (dselans) [inline function.go:245]: Add context/expand error

**Review by dselans — APPROVED** [review]

File: function.go (hunk 1 above)
Comment (blinktag) [reply function.go]: Done!

File: function.go (hunk 2 above)
Comment (blinktag) [reply function.go]: Added context

**Pull request merged**